core/src/main/java/de/danoeh/antennapod/core/sync/LockingQueueWriter.java

```diff
@@ -0,0 +1,66 @@
+package de.danoeh.antennapod.core.sync;
+
+import android.content.Context;
+
+import org.json.JSONException;
+
+import de.danoeh.antennapod.model.feed.FeedMedia;
+import de.danoeh.antennapod.net.sync.model.EpisodeAction;
+
+public class LockingQueueWriter {
```

Comment:
This cannot be moved to SynchronizationQueue because the lock also includes triggering synchronization. 
I am not very fond of that name too. What name do you propose?

Reply:
How about calling this one `SynchronizationQueue` and renaming the other to something like `SynchronizationQueueStorage`? The class `SynchronizationQueueStorage` should be protected then, so that users know that they need to use the wrapper class.